File: Backend/app/schemas/maintenance.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional
from .user import UserReference
from app.models.enums import MaintenanceType, MaintenanceStatus

class MaintenanceCreate(BaseModel):
    description: str
    asset_id: int
    user_id: int
    maintenance_type: MaintenanceType = MaintenanceType.PREVENTIVE
    scheduled_date: Optional[datetime] = None
    workorder_id: Optional[int] = None

class MaintenanceRead(BaseModel):
    id: int
    description: str
    status: MaintenanceStatus
    maintenance_type: MaintenanceType
    asset_id: int
    user_id: int
    workorder_id: Optional[int] = None
    scheduled_date: Optional[datetime] = None
    completed_date: Optional[datetime] = None
    duration_hours: Optional[float] = None
    cost: Optional[float] = None
    notes: Optional[str] = None
    created_at: datetime
    updated_at: Optional[datetime] = None
    
    class Config:
        from_attributes = True

class MaintenanceReadWithUser(MaintenanceRead):
    technician: Optional[UserReference] = None
    
    class Config:
        from_attributes = True

class MaintenanceUpdate(BaseModel):
    description: Optional[str] = None
    status: Optional[str] = None
    maintenance_type: Optional[str] = None
    user_id: Optional[int] = None
    scheduled_date: Optional[datetime] = None
    completed_date: Optional[datetime] = None
    duration_hours: Optional[float] = None
    cost: Optional[float] = None
    notes: Optional[str] = None
    workorder_id: Optional[int] = None

File: Frontend/zyro-web/src/app/types/maintenance.ts
/**
 * Tipos para mantenimiento
 */

export interface MaintenanceCreate {
  description: string;
  asset_id: number;
  user_id: number;
  maintenance_type?: string;
  scheduled_date?: string;
  workorder_id?: number;
}

export interface MaintenanceRead {
  id: number;
  description: string;
  status: string;
  maintenance_type: string;
  asset_id: number;
  user_id: number;
  workorder_id?: number;
  organization_id: number;
  scheduled_date?: string;
  completed_date?: string;
  duration_hours?: number;
  cost?: number;
  notes?: string;
  created_at: string;
  updated_at?: string;
}

export interface MaintenanceUpdate {
  description?: string;
  status?: string;
  maintenance_type?: string;
  user_id?: number;
  scheduled_date?: string;
  completed_date?: string;
  duration_hours?: number;
  cost?: number;
  notes?: string;
  workorder_id?: number;
}
